app/controllers/ppm_estimator_controller.rb

```diff
@@ -0,0 +1,21 @@
+class PpmEstimatorController < ApplicationController
+  def index
+    return lookups unless request.xhr?
+    return render plain: '', status: :not_found unless ppm_estimator.valid?
+
+    render partial: 'ppm_estimator/estimate_table'
+  end
+
+  private
+
+  # populate some of the drop downs in the form
+  def lookups
+    @entitlements ||= Entitlement.all
+    @entitlements_json ||= @entitlements.to_json
```

Comment:
You could probably dispense with this instance variable in favor of calling `@entitlements.to_json` directly in the view.

Reply:
Indeed.